### PR TITLE
Added json reporter to CLI and Documents

### PR DESCRIPTION
**Pull request by fijijavis**
## Proposed changes

Adds a new JSON reporting option to the CLI.  See Issue [#1284](https://github.com/webdriverio/webdriverio/issues/1284)

## Types of changes

What types of changes does your code introduce to WebdriverIO?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x ] I have read the [CONTRIBUTING](/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: docs/guide/reporters/json.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        }
+      ]
+    }  
+  ] 
+}
+```
+
+----
+
+For more information on WebdriverIO see the [homepage](http://webdriver.io).
```
Comment (christian-bromann) [inline docs/guide/reporters/json.md]: Can we remove that, this makes sense in your project readme but not necessary here

File: docs/guide/reporters/json.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
+WDIO JSON Reporter
```
Comment (christian-bromann) [inline docs/guide/reporters/json.md]: We need a preamble here, see how other files do that: https://raw.githubusercontent.com/webdriverio/webdriverio/master/docs/guide/reporters/dot.md. See the `index` property, make sure you order it between the last  reporter and before the custom reporter

**Comment by christian-bromann** [comment]
Two things, otherwise LGTM

**Comment by fijijavis** [comment]
Got it.  Thanks for the review

**Comment by christian-bromann** [comment]
@fijijavis sweet, one last thing, could you rebase against current master, and rearrange the index again? I added a new reporter last night as well.

**Comment by fijijavis** [comment]
I rebased but the only update was my other pull request.  Turns out teamcity and junit had the same index number.  Should be good now.

**Comment by christian-bromann** [comment]
@fijijavis mhm, something went wrong, you know have all the recent changes in this PR. I can't merge it anymore. Could you do the following: switch to master branch, pull recent changes from master, create a new branch, apply your changes, create a new PR.

Sorry but this PR got messed up.

**Comment by fijijavis** [comment]
Sure I'll give it a try in the morning

**Pull request closed without merging**